packages/cloudflare/src/cli/build/patches/plugins/fetch-cache-wait-until.ts

```diff
@@ -0,0 +1,57 @@
+import { patchCode } from "../ast/util.js";
+import type { ContentUpdater } from "./content-updater.js";
+
+/**
+ * The following Next.js code sets values in the incremental cache for fetch calls:
+ *  https://github.com/vercel/next.js/blob/e5fc495e3d4/packages/next/src/server/lib/patch-fetch.ts#L690-L728
+ *
+ * The issue here is that this promise is never awaited in the Next.js code (since in a standard node.js server
+ * the promise will eventually simply just run) but we do need to run it inside `waitUntil` (so that the worker
+ * is not killed before the promise is fully executed), without that this promise gets discarded and values
+ * don't get saved in the incremental cache.
+ *
+ * This function wraps the promise in a `waitUntil` call (retrieved from `globalThis.__openNextAls.getStore()`).
+ */
+export function patchFetchCacheSetMissingWaitUntil(updater: ContentUpdater) {
+  return updater.updateContent(
+    "patch-fetch-cache-set-missing-wait-until",
+    { filter: /\.(js|mjs|cjs|jsx|ts|tsx)$/, contentFilter: /Failed to set fetch cache/ },
```

Comment:
Using an exported function as the heuristic is probably more robust, i.e. `function createPatchedFetcher(`

Reply:
I don't think we can use something like `function createPatchedFetcher(` since the function's name gets minified

the error message text here is very nice in my opinion since it's also uneffected by minification

alternatively I could use something like `arrayBuffer().then` which should be pretty stable even after minification, but is probably more brittle and prone to false positives in my opinion

Reply:
> alternatively I could use something like arrayBuffer().then which should be pretty stable even after minification, but is probably more brittle and prone to false positives in my opinion

As I commented we can also use the file name to restrict the scope so that there are less false +

Reply:
ok, updated